uk.co.compucorp.civicrm.hrleaveandabsences/js/src/leave-absences/shared/controllers/sub-controllers/request-modal-details-leave.controller.js

```diff
@@ -396,6 +399,28 @@ define([
       timeObject.maxAmount = '0';
     }
 
+    /**
+     * Rounds work day times according to the base, if times exist.
+     *
+     * @param {Object} workDay object resolved by leaveRequest.getWorkDayForDate()
+     */
+    function roundWorkDayTimes (workDay) {
+      var unix, updatedUnix;
+      var base = sharedSettings.timeBaseInMinutes * 60;
+
+      ['from', 'to'].forEach(function (timeType) {
+        var timeKey = ['time_' + timeType];
+
+        if (!workDay[timeKey]) {
+          return;
+        }
+
+        unix = moment(workDay[timeKey], timeFormat).unix();
+        updatedUnix = Math[timeType === 'to' ? 'ceil' : 'floor'](unix / base) * base;
```

Comment:
And this one too. I see this pattern being repeated again, and again. And it's not an easy ready either.

Reply:
Done.